core/client/src/client.rs

```diff
@@ -74,10 +80,12 @@ pub struct Client<B, E, Block, RA> where Block: BlockT {
 	import_notification_sinks: Mutex<Vec<mpsc::UnboundedSender<BlockImportNotification<Block>>>>,
 	finality_notification_sinks: Mutex<Vec<mpsc::UnboundedSender<FinalityNotification<Block>>>>,
 	import_lock: Mutex<()>,
-	importing_block: RwLock<Option<Block::Hash>>, // holds the block hash currently being imported. TODO: replace this with block queue
+	// holds the block hash currently being imported. TODO: replace this with block queue
+	importing_block: RwLock<Option<Block::Hash>>,
 	block_execution_strategy: ExecutionStrategy,
 	api_execution_strategy: ExecutionStrategy,
 	_phantom: PhantomData<RA>,
+	inherent_data_providers: InherentDataProviders,
```

Comment:
TBH I'm not a fan of stuffing this into the client; it seems to me that it belongs to a higher level and the reason for putting it in client is because it is not otherwise easily available during service setup.

A temporary configuration as is used for grandpa is a better fit, because then we can replace it with proper service setup pipeline object later.